Python/crossinterp.c

```diff
@@ -2629,20 +2820,25 @@ _PyXI_Exit(_PyXI_session *session, _PyXI_errcode errcode,
 /* in an active cross-interpreter session */
 
 static void
-_capture_current_exception(_PyXI_session *session)
+_capture_current_exception(PyThreadState *tstate,
+                           _PyXI_error_override *override,
+                           _PyXI_error *error)
 {
-    assert(session->error.info == NULL);
+    assert(!xi_error_is_set(error));
     if (!PyErr_Occurred()) {
-        assert(session->error.override == NULL);
+        assert(error->override == NULL);
+        xi_error_set_override(error, override);
         return;
     }
 
     // Handle the exception override.
-    _PyXI_errcode *override = session->error.override;
-    session->error.override = NULL;
-    _PyXI_errcode errcode = override != NULL
-        ? *override
-        : _PyXI_ERR_UNCAUGHT_EXCEPTION;
+    _PyXI_error_override _override = {
+        .code = _PyXI_ERR_UNCAUGHT_EXCEPTION,
+    };
+    if (override == NULL) {
+        override = &_override;
+    }
+    _PyXI_errcode errcode = override->code;
```

Comment:
For `test_running`, putting `_override.code = errcode;` in the `_PyXI_ERR_ALREADY_RUNNING` case (several lines below) works well for me.

Reply:
fixed